From 62d0b5c1689076468e95b5473bffd599b51f8d8e Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Mon, 29 Sep 2025 14:14:44 +0800
Subject: [PATCH 07/16] add src/include/override/sys/prctl.h to avoid
 redefinition error

The linux/prctl.h has all the needed macros, what we need is the
prctl function definition. This avoids error of redefinition.

Upstream-Status: Inappropriate [musl specific]

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 src/include/override/sys/prctl.h | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)
 create mode 100644 src/include/override/sys/prctl.h

diff --git a/src/include/override/sys/prctl.h b/src/include/override/sys/prctl.h
new file mode 100644
index 0000000000..20468bd30f
--- /dev/null
+++ b/src/include/override/sys/prctl.h
@@ -0,0 +1,17 @@
+#ifndef _SYS_PRCTL_H
+#define _SYS_PRCTL_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <linux/prctl.h>
+
+int prctl (int, ...);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
-- 
2.34.1

